src/Common/Services/Mail/MailConfiguration.cs

```diff
@@ -2,5 +2,13 @@ namespace Passwordless.Common.Services.Mail;
 
 public class MailConfiguration
 {
+    /// <summary>
+    /// The default email address to use as the sender.
+    /// </summary>
     public string? From { get; set; }
+
+    /// <summary>
+    /// The ordered list of mail providers to use.
+    /// </summary>
+    public List<BaseMailProviderOptions> Providers { get; set; } = new();
```

Comment:
I think it should be `IReadOnlyList<>` to avoid accidental mutations after the configuration has been initialized. In `AddMail()` you can do: `var providers = new List<...>();` and then `o.Providers = providers`.